### ISSUE TITLE
[HOLD for 42143] [$2000] [LOW] IOS - Keypad appears for a moment in details page on saving Private notes

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#35779]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):** applausetester+cknew@applause.expensifail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

## Action Performed:
Precondition: User has not saved Private notes
1. Open a workspace #announce room
2. Tap header link to go details page
3. Select Private notes
4. Enter notes & tap on Save button

## Expected Result:
Keypad should not appear in details or settings page while saving Private notes

## Actual Result:
Keypad appears for a moment in details/settings page while saving Private notes

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/92cf7644-89a5-462f-b5bc-242f2db291ff

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d02b0565c21176af</li>
        <li>Upwork Job ID: 1754200247556661248</li>
        <li>Last Price Increase: 2024-03-25</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #vip-vsp
CC @quinthar

**Comment by getusha** [comment]
Looks like a very minor issue, @bfitzexpensify i think we can close it, wdyt?

**Comment by bfitzexpensify** [comment]
@getusha Agreed that it's very minor, but assuming there is a clear and easy solution, we should address this. What do you think of @mkhutornyi's proposal?

**Comment by getusha** [comment]
Thanks! will test @mkhutornyi's proposal.

**Comment by getusha** [comment]
@mkhutornyi are you able to reproduce this issue?

https://github.com/Expensify/App/assets/75031127/d811ed93-8b8e-4579-bde5-50c2c672a7b2

**Comment by getusha** [comment]
Still not able to repro, what's the version your iOS version?

**Comment by getusha** [comment]
asked for other c+
https://expensify.slack.com/archives/C02NK2DQWUX/p1707408881410369

**Comment by situchan** [comment]
@getusha this is reproducible not only on this page but also other pages which have multiline input

**Comment by situchan** [comment]
The proposed solution doesn't fix issue completely

**Comment by situchan** [comment]
Another reproducible case: (swipe left to go back)


https://github.com/Expensify/App/assets/108292595/0ec5e69f-3b63-4209-9713-e0589b238b6b

**Comment by situchan** [comment]
Please let me know if it's still not reproducible for you. If still, I can take over.

**Comment by getusha** [comment]
@situchan I missed your comment there! @mananjadhav offered to take over.

I'm still unable to reproduce

https://github.com/Expensify/App/assets/75031127/90b0160b-de89-4576-b7a1-3f722a3c40d0

**Comment by mananjadhav** [comment]
I am able to reproduce. Will check the proposal in a while.

**Comment by bfitzexpensify** [comment]
Were you able to review the proposal @mananjadhav?

**Comment by mananjadhav** [comment]
Based on the previous [comment](https://github.com/Expensify/App/issues/35779#issuecomment-1934477202) it looks like the proposal doesn't work. Open for proposals.

**Comment by bfitzexpensify** [comment]
No proposals yet - will leave this open for another week, but if we get no traction, I'll raise it in Slack before we double the price given how minor it is.

**Comment by bfitzexpensify** [comment]
Still awaiting proposals

**Comment by bfitzexpensify** [comment]
Same update - still awaiting proposals

**Comment by bfitzexpensify** [comment]
Doubling this - I think the behaviour in https://github.com/Expensify/App/issues/35779#issuecomment-1934479361 is worse than the scenario in the OP and makes this something worth addressing.

**Comment by FitseTLT** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
IOS - Keypad appears for a moment in details page on saving Private notes
### What is the root cause of that problem?
We are dismissing the keyboard and immediately navigating without waiting for the dismissing of the keyboard here 
https://github.com/Expensify/App/blob/29ad8787c7c889271c117dd4420109f9be8e180d/src/pages/PrivateNotes/PrivateNotesEditPage.tsx#L103-L108
### What alternative solutions did you explore? (Optional)
We need to navigate under `runAfterInteraction`, change it to 
```
Keyboard.dismiss();
        InteractionManager.runAfterInteractions(() => {
            if (!Object.values<Note>({...report.privateNotes, [route.params.accountID]: {note: editedNote}}).some((item) => item.note)) {
                ReportUtils.navigateToDetailsPage(report);
            } else {
                Navigation.goBack(ROUTES.PRIVATE_NOTES_LIST.getRoute(report.reportID));
            }
        });
```
But to avoid unnecessary delay of navigation where there is no on screen keyboards (like for web and desktop) we will only use it only if `canUseTouchScreen` is true

**Comment by bfitzexpensify** [comment]
How does the proposal in https://github.com/Expensify/App/issues/35779#issuecomment-1961326402 look @mananjadhav?

**Comment by mananjadhav** [comment]
@FitseTLT's proposal looks promising, but I'll test it out and confirm.

**Comment by mananjadhav** [comment]
@mkhutornyi which PR are we talking about here? @ntdiary did you get a chance to look at the previous comment?

**Comment by bfitzexpensify** [comment]
Where are we at with this one @mananjadhav? Sounds like we're still awaiting a proposal that doesn't use un-ideal workarounds?

**Comment by FitseTLT** [comment]
> Where are we at with this one @mananjadhav? Sounds like we're still awaiting a proposal that doesn't use un-ideal workarounds?

Considering we are solving platform-specific issue, I don't think my solution is a workaround as the problem is being caused by immediately calling navigate after `Keyboard.dismiss` call and we have used the same solution to solve similar `mWeb` issue [here](https://github.com/Expensify/App/pull/32901) but lets wait for @mananjadhav thoughts 👍

**Comment by mananjadhav** [comment]
Thanks @mkhutornyi @FitseTLT for keeping the discussion active. @FitseTLT can you confirm if [this scenario](https://github.com/Expensify/App/issues/35779#issuecomment-1934479361) is addressed.

**Comment by bfitzexpensify** [comment]
Bump here @FitseTLT - thank you!

**Comment by bfitzexpensify** [comment]
Another bump @FitseTLT

**Comment by bfitzexpensify** [comment]
Still open for proposals (though also waiting on a clarification from @FitseTLT on their proposal to see if we can move forward with it).

**Comment by bfitzexpensify** [comment]
Same update again - still waiting on proposals.

**Comment by bfitzexpensify** [comment]
Still open for proposals.

**Comment by bfitzexpensify** [comment]
I updated the price to reflect the lack of new proposals here over the last couple of weeks. But happy to put it on hold for https://github.com/Expensify/App/pull/38232.

**Comment by ikevin127** [comment]
We might not need to hold since it looks like the issue is not reproducible anymore.
Can we ask for a re-test to confirm ?

I tested on dev with latest main branch, both real device and simulator:

<details>
<summary>Videos</summary>

| Real device | Simulator |
| ------------- | ------------- |
| <video src="https://github.com/Expensify/App/assets/56457735/802b03e0-5859-472c-bc28-e62b762e7027"> | <video src="https://github.com/Expensify/App/assets/56457735/72096e8e-0847-44b5-96dc-dfb92bd9c867"> |

</details>

**Comment by bfitzexpensify** [comment]
Still holding on https://github.com/Expensify/App/pull/38232

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Same update - still held

**Comment by bfitzexpensify** [comment]
https://github.com/Expensify/App/pull/38232 is held until the week of 19/04 - moving this to weekly and then will adjust back once there's progress there again

**Comment by bfitzexpensify** [comment]
Still held on https://github.com/Expensify/App/pull/38232

**Comment by bfitzexpensify** [comment]
Same status, still held.

**Comment by bfitzexpensify** [comment]
Still held.

**Comment by bfitzexpensify** [comment]
https://github.com/Expensify/App/pull/38232 is closed, moved to https://github.com/Expensify/App/pull/42143.

**Comment by bfitzexpensify** [comment]
We're still held, updated title.

**Comment by bfitzexpensify** [comment]
Remains held.

**Comment by bfitzexpensify** [comment]
Remains held.

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held on https://github.com/Expensify/App/pull/42143.

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Same update, still held

**Comment by bfitzexpensify** [comment]
Remains held

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held on https://github.com/Expensify/App/pull/42143.

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held. Moving this to monthly for the moment

**Comment by bfitzexpensify** [comment]
Remains held

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held on https://github.com/Expensify/App/pull/42143.

**Comment by kavimuru** [comment]
Bug is fixed 

https://github.com/user-attachments/assets/29964ba9-3a0e-4e2f-a725-b41eeb4ecb61

**Comment by bfitzexpensify** [comment]
Awesome, let's close this out.

**Comment by bfitzexpensify** [comment]
Given it's no longer a reproducible bug, I don't see why we would reopen this.
